Core/src/org/sleuthkit/autopsy/modules/pictureanalysis/impls/HEICProcessor.java

```diff
@@ -0,0 +1,245 @@
+        final String moduleOutputDirectory = Case.getCurrentCaseThrows().getModuleDirectory();
+
+        return Paths.get(moduleOutputDirectory,
+                HEIC_MODULE_FOLDER,
+                String.valueOf(file.getId()));
+    }
+
+    /**
+     * Create any sub directories within the module output folder.
+     */
+    private void createModuleOutputFolder(AbstractFile file) throws IOException, NoCurrentCaseException {
+        final Path moduleOutputFolder = getModuleOutputFolder(file);
+
+        if (!Files.exists(moduleOutputFolder)) {
+            Files.createDirectories(moduleOutputFolder);
+        }
+    }
+
+    @Override
+    public void process(IngestJobContext context, AbstractFile file) {
+        try {
+            if (IMAGE_MAGICK_PATH == null) {
+                return;
+            }
+            createModuleOutputFolder(file);
+
+            if (context.fileIngestIsCancelled()) {
+                return;
+            }
+
+            final Path localDiskCopy = extractToDisk(file);
+
+            convertToJPEG(context, localDiskCopy, file);
+        } catch (IOException ex) {
+            logger.log(Level.WARNING, "I/O error encountered during HEIC photo processing.", ex);
+        } catch (TskCoreException ex) {
+            logger.log(Level.SEVERE, "Unable to add pictures as derived files.", ex);
+        } catch (NoCurrentCaseException ex) {
+            logger.log(Level.WARNING, "No open case!", ex);
+        }
+    }
+
+    /**
+     * Copies the HEIC container to disk in order to run ImageMagick.
+     */
+    private Path extractToDisk(AbstractFile heicFile) throws IOException {
+        final String tempDir = Case.getCurrentCase().getTempDirectory();
+        final String heicFileName = FileUtil.escapeFileName(heicFile.getName());
+
+        final Path localDiskCopy = Paths.get(tempDir, heicFileName);
+
+        try (BufferedInputStream heicInputStream = new BufferedInputStream(new ReadContentInputStream(heicFile))) {
+            Files.copy(heicInputStream, localDiskCopy, StandardCopyOption.REPLACE_EXISTING);
+            return localDiskCopy;
+        }
+    }
+
+    private void convertToJPEG(IngestJobContext context, Path localDiskCopy,
+            AbstractFile heicFile) throws IOException, TskCoreException, NoCurrentCaseException {
+
+        // First step, run ImageMagick against this heic container.
+        final Path moduleOutputFolder = getModuleOutputFolder(heicFile);
+
+        final String baseFileName = FilenameUtils.getBaseName(FileUtil.escapeFileName(heicFile.getName()));
+        final Path outputFile = moduleOutputFolder.resolve(baseFileName + ".jpg");
+
+        // ImageMagick will write the primary image to the output file.
+        // Any additional images found within the HEIC container will be
+        // formatted as fileName-1.jpg, fileName-2.jpg, etc.
+        final ProcessBuilder processBuilder = new ProcessBuilder()
+                .command(IMAGE_MAGICK_PATH.toString(),
```

Comment:
Could you add a redirectError to processBuilder so that it will redirect the error to the temp directory for the file.  This will make debugging easier in the future when issues happen.